.github/workflows/update-kernels.yml

```
@@ -21,13 +42,35 @@ jobs:
         with:
           ref: kernels
 
+      - name: Setup jq
+        run: apt-get update && apt-get install jq -y
+
       - name: Run crawler for x86_64
         run: |
-          kernel-crawler crawl --distro=* > x86_64/list.json
+          mkdir $RUNNER_TEMP/x86_64
+          kernel-crawler crawl --distro=${{ inputs.distro }} > $RUNNER_TEMP/x86_64/list.json
 
       - name: Run crawler for aarch64
         run: |
-          kernel-crawler crawl --distro=* --arch=aarch64 > aarch64/list.json
+          mkdir $RUNNER_TEMP/aarch64
+          kernel-crawler crawl --distro=${{ inputs.distro }} --arch=aarch64 > $RUNNER_TEMP/aarch64/list.json
+
+      - name: Single distro update
+        if: github.event_name == 'workflow_dispatch' && ${{ inputs.distro }} != '*'
+        run: |
+          jq --arg distroKey "${{ inputs.distro }}" \
+          --slurpfile newValues $RUNNER_TEMP/x86_64/list.json \
+          'if .[$distroKey] then .distroKey = $newValues else . end' \
+          x86_64/list.json > $RUNNER_TEMP/x86_64/output.json
+          jq --arg distroKey "${{ inputs.distro }}" \
+          --slurpfile newValues $RUNNER_TEMP/aarch64/list.json \
+          'if .[$distroKey] then .distroKey = $newValues else . end' \
+          aarch64/list.json > $RUNNER_TEMP/aarch64/output.json
+
+      - name: Update json lists
+        run: |
+          mv $RUNNER_TEMP/x86_64/output.json x86_64/list.json
+          mv $RUNNER_TEMP/aarch64/output.json aarch64/list.json
```

```suggestion
          mv $RUNNER_TEMP/x86_64/list.json x86_64/list.json
          mv $RUNNER_TEMP/aarch64/list.json aarch64/list.json
```

Since it is how they're created above :)